### PR TITLE
Exporter: add memos export

### PR DESCRIPTION
**Pull request by fierceX**
Add an export method for memos, which originates from [flomo](https://flomoapp.com/) and now has an open-source self-deployment solution:[memos](https://github.com/usememos/memos)

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10411)

### CODE REVIEW

File: plugins/exporter.koplugin/target/memos.lua [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                keep_menu_open = true,
+                callback = function()
+                    local auth_dialog
+                    auth_dialog = InputDialog:new {
+                        title = _("Set API URL for Memos"),
+                        input = self.settings.api,
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
```
Comment (hius07) [inline plugins/exporter.koplugin/target/memos.lua:65]: Can you add `id = "close"` to enable exiting the dialog on non-touch devices.

Comment (fierceX) [reply plugins/exporter.koplugin/target/memos.lua:65]: I am not familiar with lua and koreader, this is what I changed according to the code of other export plugins, now please see if there is any problem with my modification, thank you.

File: plugins/exporter.koplugin/main.lua [hunk 2]
```diff
@@ -38,7 +38,7 @@ local _ = require("gettext")
 
 -- migrate settings from old "evernote.koplugin" or from previous (monolithic) "exporter.koplugin"
 local function migrateSettings()
-    local formats = { "html", "joplin", "json", "readwise", "text", "my_clippings" }
+    local formats = { "html", "joplin", "json", "readwise", "text", "my_clippings", "memos" }
```
Comment (pazos) [inline plugins/exporter.koplugin/main.lua:41]: It used to be sorted.

Could you please sort it again (both memos and my_clippings). Thanks!

File: plugins/exporter.koplugin/main.lua [hunk 3]
```diff
@@ -57,6 +57,7 @@ local function migrateSettings()
         new_settings["joplin"].port = settings.joplin_port
         new_settings["joplin"].token = settings.joplin_token
         new_settings["readwise"].token = settings.readwise_token
+        new_settings["memos"].api = settings.memos_api
```
Comment (pazos) [inline plugins/exporter.koplugin/main.lua:60]: It isn't required. It is just for previous iteration of the plugin. Please remove

File: plugins/exporter.koplugin/main.lua (hunk 2 above)
Comment (fierceX) [reply plugins/exporter.koplugin/main.lua:41]: Is that the correct ordering: `'html', 'joplin', 'json', 'memos', 'my_clippings', 'readwise', 'text'` ?

**Comment by fierceX** [comment]
@hius07 @pazos Hi, what other changes do I need to make？

**Comment by hius07** [comment]
Looks good (didn't tested the memos service).

**Comment by fierceX** [comment]
@hius07 Do I need to add tests to the code. I tested it myself and it works. Do I need to provide screenshots or something like that?

**Comment by hius07** [comment]
> I tested it myself and it works.

Good, it's enough I think.

**Review by hius07 — APPROVED** [review]

**Comment by cpuks** [comment]
Anyone got this working? As I've tried 2024.3.1/.3/.1 and 2023.10 - no luck although API works with curl and android memos app locally and via FQDN, flomo has same issue too - exporter says successful but no data retrieved in apps. Also once I manually specify memos via share option I got crash every time no matter what. 

`--------- beginning of system
03-05 23:05:22.617 I/ForceDarkHelperStubImpl(32612): setViewRootImplForceDark: false for org.koreader.launcher.CrashReportActivity@f0c12a4, reason: AppDarkModeEnable
03-05 23:23:56.433 W/Looper  (32612): PerfMonitor doFrame : time=769ms vsyncFrame=0 latency=0ms procState=-1
`

**Comment by fierceX** [comment]
@cpuks sorry it's been a busy time, can you tell me the version of memos you are currently using, I'll test it out

**Pull request merged**